### PR TITLE
Fix memory leaks in RBin reported by Coverity

### PR DESCRIPTION
**Pull request by trufae**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

This PR fixes a bunch of memory leaks and uninitialized variable usage in dex and pe

**Test plan**

Wait for the CI

**Closing issues**

None

### CODE REVIEW

File: libr/bin/p/bin_dex.c [hunk 1]
```diff
@@ -752,7 +727,9 @@ static void dex_parse_debug_item(RBinFile *bf, RBinDexClass *c, int MI, int MA,
 			}
 		}
 	}
-	free (debug_positions);
+beach:
+	r_list_free (debug_positions);
+	// free (debug_positions);
```
Comment (ret2libc) [inline libr/bin/p/bin_dex.c:732]: why?

Comment (trufae) [reply libr/bin/p/bin_dex.c]: leftover

File: libr/bin/p/bin_dex.c [hunk 2]
```diff
@@ -752,7 +727,8 @@ static void dex_parse_debug_item(RBinFile *bf, RBinDexClass *c, int MI, int MA,
 			}
 		}
 	}
-	free (debug_positions);
+beach:
+	r_list_free (debug_positions);
 	free (debug_locals);
 	free (emitted_debug_locals);
```
Comment (ret2libc) [inline libr/bin/p/bin_dex.c:733]: this seems to be a list as well, we should free it with `r_list_free`.

**Comment by ret2libc** [comment]
@XVilka bsd ci failing again :/ anyway, this is ok for me.

**Pull request merged**